File: src/com/dndsuite/core/gameobjects/Zombie.java
package com.dndsuite.core.gameobjects;

import com.dndsuite.core.tasks.Attack;
import com.dndsuite.core.tasks.Task;
import com.dndsuite.maths.Vector;

/**
 * This class represents a Zombie game piece
 * 
 * @author calvi
 *
 */
public class Zombie extends Entity {

	public Zombie(Vector pos, Vector rot) {
		super("Zombie", pos, rot);

		experience = 50;
		level = 3;
		abilityScores[Entity.STR] = 13;
		abilityScores[Entity.DEX] = 6;
		abilityScores[Entity.CON] = 16;
		abilityScores[Entity.INT] = 3;
		abilityScores[Entity.WIS] = 6;
		abilityScores[Entity.CHA] = 5;
		baseAbilityScores[Entity.STR] = 13;
		baseAbilityScores[Entity.DEX] = 6;
		baseAbilityScores[Entity.CON] = 16;
		baseAbilityScores[Entity.INT] = 3;
		baseAbilityScores[Entity.WIS] = 6;
		baseAbilityScores[Entity.CHA] = 5;
		generateHealth(8);

		Task task;
		task = new Attack(this, 3);
		addBaseTask(task);

		getTasks(); // populate initial availableTasks list
	}

}

File: scripts/entities/zombie.lua
function define()
	DamageType = luajava.bindClass("com.dndsuite.dnd.data.DamageType")
	Entity = luajava.bindClass("com.dndsuite.core.gameobjects.Entity")
	
	-- Set name
	self:setName("Zombie")
	
	-- Apply all relevant tags
	self:addTag(Entity.UNDEAD)
	
	-- Set challenge data
	self:setLevel(3)
	self:setExperience(50)
	
	-- Set health data
	self:setHealthBase(22)
	self:setHealthMax(22)
	self:setHealth(22)
	self:setHealthTmp(0)
	
	-- Populate inventory
	
	-- Add base tasks
	task = luajava.newInstance("com.dndsuite.core.tasks.Task", "scripts/tasks/zombie_slam.lua")
	self:addBaseTask(task)
	task = luajava.newInstance("com.dndsuite.core.tasks.Attack", self)
	self:addBaseTask(task)
	
	-- Add active effects
	effect = luajava.newInstance("com.dndsuite.core.effects.Effect", "scripts/effects/undead_fortitude.lua", self, self)
	self:addEffect(effect)
	effect = luajava.newInstance("com.dndsuite.core.effects.DamageImmunity", self, self, DamageType.POISON)
	self:addEffect(effect)
	effect = luajava.newInstance("com.dndsuite.core.effects.SaveProficiency", self, self, Entity.WIS)
	self:addEffect(effect)
	
	-- Add item proficiencies
	
	-- Set held and worn equipment
	
	-- Set ability scores
	self:setBaseAbilityScore(Entity.STR, 13)
	self:setBaseAbilityScore(Entity.DEX, 6)
	self:setBaseAbilityScore(Entity.CON, 16)
	self:setBaseAbilityScore(Entity.INT, 3)
	self:setBaseAbilityScore(Entity.WIS, 6)
	self:setBaseAbilityScore(Entity.CHA, 5)
end